src/admin/components/ResourceTabs/PolygonReviewTab/components/PolygonStatus/StatusDisplay .tsx

```diff
@@ -0,0 +1,249 @@
+  Project: menuProjectOptions
+};
+
+const DescriptionStatusMap = {
+  Polygon: "Are you sure you want to change the polygon status to",
+  Site: "Are you sure you want to change the site status to",
+  Project: "Are you sure you want to change the project status to"
+};
+
+const DescriptionRequestMap = {
+  Polygon: "Provide an explanation for your change request for the polygon",
+  Site: "Provide an explanation for your change request for the site",
+  Project: "Provide an explanation for your change request for the project"
+};
+
+const StatusDisplay = ({ titleStatus = "Polygon", mutate, refresh, name, record, setSelectedPolygon }: StatusProps) => {
+  const { refetch: reloadEntity } = useShowContext();
+  const [notificationStatus, setNotificationStatus] = useState<{
+    open: boolean;
+    message: string;
+    type: "success" | "error" | "warning";
+    title: string;
+  }>({
+    open: false,
+    message: "",
+    type: "success",
+    title: "Success!"
+  });
+
+  const { openModal, closeModal } = useModalContext();
+  const contentStatus = (
+    <Text variant="text-12-light" as="p" className="text-center">
+      {DescriptionStatusMap[titleStatus]} <b style={{ fontSize: "inherit" }}>{name}</b>?
+    </Text>
+  );
+  const contentRequest = (
+    <Text variant="text-12-light" as="p" className="text-center">
+      {DescriptionRequestMap[titleStatus]} <b style={{ fontSize: "inherit" }}>{name}</b>?
+    </Text>
+  );
+
+  const openFormModalHandlerStatus = () => {
+    openModal(
+      <ModalConfirm
+        title={`${titleStatus} Status Change`}
+        commentArea
+        menuLabel={""}
+        menu={menuOptionsMap[titleStatus]}
+        onClose={closeModal}
+        content={contentStatus}
+        onConfirm={async (text: any, opt) => {
+          const option = menuOptionsMap[titleStatus].find(option => option.value === opt[0]);
+          let response;
+          try {
+            response = await mutate({
+              pathParams: { uuid: record?.uuid },
+              body: {
+                status: option?.status,
+                comment: text,
+                type: "status"
+              }
+            });
+            if (response.poly_id) {
+              setSelectedPolygon(response?.poly_id);
+            }
+            setNotificationStatus({
+              open: true,
+              message: "Your Status Update was just saved!",
+              type: "success",
+              title: "Success!"
+            });
+            setTimeout(() => {
+              setNotificationStatus({
+                open: false,
+                message: "",
+                type: "success",
+                title: "Success!"
+              });
+            }, 3000);
+          } catch (e) {
+            alert("The request encountered an issue, or the comment exceeds 255 characters.");
+            console.error(e);
+          } finally {
+            refresh();
+            reloadEntity();
+            closeModal;
```

Comment:
This looks like it's supposed to be a function call? `closeModal();`

Reply:
Thanks for pointing that out! I've updated the code to ensure closeModal is called as a function.